src/main/java/uk/nhs/digital/nhsconnect/lab/results/inbound/RecepProducerService.java

```diff
@@ -0,0 +1,151 @@
+package uk.nhs.digital.nhsconnect.lab.results.inbound;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Interchange;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.InterchangeTrailer;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.MessageTrailer;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.RecepBeginningOfMessage;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.RecepHeader;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.RecepMessageDateTime;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.RecepMessageHeader;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.RecepNationalHealthBody;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ReferenceInterchangeRecep;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ReferenceMessageRecep;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Segment;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.lab.results.sequence.SequenceService;
+import uk.nhs.digital.nhsconnect.lab.results.utils.TimestampService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class RecepProducerService {
+
+    private final SequenceService sequenceService;
+    private final TimestampService timestampService;
+
+    public String produceRecep(final Interchange receivedInterchangeFromHa) throws EdifactValidationException {
+        final var segments = mapEdifactToRecep(receivedInterchangeFromHa);
+        return toEdifact(segments);
+    }
+
+    private String toEdifact(final List<Segment> segments) {
+        return segments.stream()
+            .map(Segment::toEdifact)
+            .collect(Collectors.joining("\n"));
+    }
+
+    private List<Segment> mapEdifactToRecep(final Interchange receivedInterchange) throws EdifactValidationException {
```

Comment:
Do we know the contents of the edifact/recep yet? As I noticed we're mapping GP names and addresses, etc

Reply:
From discussion with @bartek-sarul we agreed to use the same stuff as in NHAIS for now, and then if we find out something is meant to be different we'll change it then. The expectation at the moment is apparently that they are indeed most likely going to be the same.

Reply:
Ok, that's fair enough